File: LeetCode/Math/Rust/Leetcode793-preimage-size-of-factorial-zeroes-function.rs
impl Solution {
    pub fn preimage_size_fzf(k: i32) -> i32 {
        let f = |mut x : i64| -> i64 {
            let mut res = 0;
            while x > 0 {
                res += x / 5;
                x /= 5;
            }
            res
        };

        let calcu = |k:i32| -> i64 {
            let (mut l,mut r) = (-1 as i64, 1e18 as i64);
            while l < r {
                let mid:i64 = (l + r + 1) >> 1;
                if f(mid) <= (k as i64) {
                    l = mid;
                }else {
                    r = mid - 1;
                }
            }
            r
        };

        (calcu(k) - calcu(k - 1)) as i32
    }
}

File: LeetCode/Math/Python/Leetcode793-preimage-size-of-factorial-zeroes-function.py
class Solution:
    def preimageSizeFZF(self, k: int) -> int:
        def f(x):
            res = 0
            while x:
                res += (x // 5)
                x //= 5
            return res

        def calcu(k):
            l, r = -1, int(1e18)
            while l < r:
                mid = (l + r + 1) >> 1
                if f(mid) <= k: l = mid
                else: r = mid - 1
            return r

        return calcu(k) - calcu(k - 1)
